src/main/java/com/iridium/iridiumskyblock/listeners/EntityDamageListener.java

```diff
@@ -3,46 +3,173 @@
+                if (hasNoCooldown(player)) {
+                    player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMobs.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                }
+            }
+        }
+    }
+
+    public void handleDamageBetweenPlayers(EntityDamageByEntityEvent event, Player attacker, Player victim, Island island) {
+        Configuration configuration = IridiumSkyblock.getInstance().getConfiguration();
+
+        if (!configuration.pvpSettings.pvpOnIslands) {
+
+            if (hasNoCooldown(attacker)) {
+                attacker.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtPlayer.replace("%prefix%", configuration.prefix)));
+            }
+
+            if (event.getCause().equals(EntityDamageEvent.DamageCause.PROJECTILE)) event.getDamager().remove();
+            event.setCancelled(true);
+            return;
+        }
+
+        User attackerUser = IridiumSkyblock.getInstance().getUserManager().getUser(attacker);
+        User victimUser = IridiumSkyblock.getInstance().getUserManager().getUser(victim);
+        Optional<IslandTrusted> attackerTrusted = IridiumSkyblock.getInstance().getIslandManager().getIslandTrusted(island, attackerUser);
+        Optional<IslandTrusted> victimTrusted = IridiumSkyblock.getInstance().getIslandManager().getIslandTrusted(island, victimUser);
+        boolean attackerIsMember = island.equals(attackerUser.getIsland().orElse(null)) || attackerTrusted.isPresent();
+        boolean victimIsMember = island.equals(victimUser.getIsland().orElse(null)) || victimTrusted.isPresent();
+
+        if (!configuration.pvpSettings.pvpBetweenMembers && attackerIsMember && victimIsMember) {
+
+            if (hasNoCooldown(attacker)) {
+                attacker.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotHurtMember.replace("%prefix%", configuration.prefix)));
             }
+
+            if (event.getCause().equals(EntityDamageEvent.DamageCause.PROJECTILE)) event.getDamager().remove();
+            event.setCancelled(true);
+        }
+    }
+
+    public void handPlayerDamage(EntityDamageEvent event, Island island) {
+        Player player = (Player) event.getEntity();
+        User user = IridiumSkyblock.getInstance().getUserManager().getUser(player);
+        Optional<IslandTrusted> trusted = IridiumSkyblock.getInstance().getIslandManager().getIslandTrusted(island, user);
+        boolean isMember = island.equals(user.getIsland().orElse(null)) || trusted.isPresent();
+
+        if ((isMember && membersExcludedDamages.contains(event.getCause())) ||
+                (!isMember && visitorsExcludedDamages.contains(event.getCause()))) {
+
+            event.setCancelled(true);
         }
+
+    }
+
+    private boolean hasNoCooldown(Player player) {
```

Comment:
The method has some side effects that the name of the method does not explain. Add Javadocs or find a better name.